From 7fa6e867892cea778333c95e209feb20cebd250e Mon Sep 17 00:00:00 2001
From: Disconnect3d <dominik.b.czarnota@gmail.com>
Date: Mon, 13 Apr 2020 16:14:10 +0200
Subject: [PATCH 207/389] Fix off by one when checking "/dev/mapper/" path

This PR fixes an off by one in `strncmp(dev->bid_name, "/dev/mapper/", 11)` check. The `"/dev/mapper/"` string literal has a length of 12 and without this fix paths like `"/dev/mapperSOMETHING"` would also be accepted.
---
 libblkid/src/devname.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libblkid/src/devname.c b/libblkid/src/devname.c
index 014b3f4..bb0211d 100644
--- a/libblkid/src/devname.c
+++ b/libblkid/src/devname.c
@@ -258,7 +258,7 @@ set_pri:
 	if (dev) {
 		if (pri)
 			dev->bid_pri = pri;
-		else if (!strncmp(dev->bid_name, "/dev/mapper/", 11)) {
+		else if (!strncmp(dev->bid_name, "/dev/mapper/", 12)) {
 			dev->bid_pri = BLKID_PRI_DM;
 			if (is_dm_leaf(ptname))
 				dev->bid_pri += 5;
-- 
1.8.3.1

